src/script/lua_api/l_object.cpp

```diff
@@ -1720,6 +1720,45 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// set_time_offset(self, offset)
+int ObjectRef::l_set_time_offset(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == NULL)
+		return 0;
+
+	int offset_i = luaL_checknumber(L, 2);
+	u16 offset = offset_i;
+	//Negative values
```

Comment:
why is the offset not just a signed number (`s16`)?

Reply:
Because timeofday is an unsigned number and when I add a negative offset to timeofday and timeofday<0 then it would be the wrong time.

Reply:
either way, an offset of, say, 23999 would effectively work the same, and if anything you could perform a modulus (t % 24000)